internal/operator/clusterutilities.go

```diff
@@ -288,14 +289,36 @@ func GetPgbackrestEnvVars(cluster *crv1.Pgcluster, depName, port string) string
 
 // GetPgbackrestBootstrapEnvVars returns a string containing the pgBackRest environment variables
 // for a bootstrap job
-func GetPgbackrestBootstrapEnvVars(restoreClusterName, depName string,
+func GetPgbackrestBootstrapEnvVars(cluster *crv1.Pgcluster,
 	restoreFromSecret *v1.Secret) (string, error) {
+	ctx := context.TODO()
+
+	depName := cluster.GetAnnotations()[config.ANNOTATION_CURRENT_PRIMARY]
+	restoreClusterName := cluster.Spec.PGDataSource.RestoreFrom
+
+	// get the namespace for the repo we're restoring from
+	restoreFromNamespace := GetBootstrapNamespace(cluster)
+
+	// Lookup an existing Service to determine its fully qualified domain name.
+	// This is inexpensive because the "net" package uses OS-level DNS caching.
+	// - https://golang.org/issue/24796
+	api := "kubernetes.default.svc"
+	cname, err := net.DefaultResolver.LookupCNAME(ctx, api)
+	if err != nil {
+		return "", err
+	}
+	domain := strings.TrimPrefix(cname, api+".")
+
+	// use DNS for the host for bootstrap Jobs since we might be bootstrapping across namespaces
+	repoHost := fmt.Sprintf(util.BackrestRepoDeploymentName, restoreClusterName) + "." +
+		restoreFromNamespace + ".svc." + domain
```

Comment:
Why do we have to do this? Why can't we use the common namespace pattern for names?

Reply:
This is to account for potential customization to the service DNS domain (for instance, if it is changed from the default of `cluster.local` via the `--service-dns-domain` when boostrapping a cluster via `kubeadm`).

Reply:
OK.